### PR TITLE
Add mypy (not strict)

### PR DESCRIPTION
**Pull request by drubinstein**
This PR adds a mypy check but not strict meaning that it only checks what has already been added. I fixed up some typing along the way.

### CODE REVIEW

File: mirdata/datasets/tonality_classicaldb.py [hunk 1]
```diff
@@ -141,19 +141,19 @@ def __init__(self, track_id, data_home):
         self.title = self.audio_path.replace(".wav", "").split("/")[-1]
 
     @core.cached_property
-    def key(self) -> str:
+    def key(self) -> Optional[str]:
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonality_classicaldb.py:144]: Hey Dave, thanks for this! I have a naive/dumb question: `no-strict` means that `mypy` doesn't check the use of `None`, or am I confused? If that's what it does, why to use `Optional` here?

Comment (drubinstein) [reply mirdata/datasets/tonality_classicaldb.py:144]: It doesnt check the use of None. Mypy does not run at runtime. It analyzes the types passed in the editor to ensure inputs/outputs match. In this case the Use of Optional says that `key` will either return `str` or `None` (`Optional` is really `Union[str, None]`)

[strict](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-strict) in this case would apply a bunch of checks which are hard to run until types are spread to the rest of the code base.

`load_key(self.key_path)` returns an `Optional[str]` (courtesy of the coerce_to_string_io decorator). The typing was previously correct so this addition fixes the typing. Thanks to mypy I found the problem in the first place.

Comment (magdalenafuentes) [reply mirdata/datasets/tonality_classicaldb.py:144]: I see, thanks @drubinstein!

**Review by rabitt — APPROVED** [review]
This is great! Thanks @drubinstein

**Comment by rabitt** [comment]
@drubinstein feel free to merge at your convenience

**Pull request merged**